PennMobile/Dining/Controllers/Statistics/DiningInsightsViewController.swift

```diff
@@ -30,17 +30,21 @@ class DiningInsightsViewController: GenericViewController {
         
         // Attempt to fetch dining insights
         DiningAPI.instance.fetchDiningInsights { (result) in
-            DispatchQueue.main.async {
-                do {
-                    self.diningInsights = try result.get()
-                }
-                catch {
-                    // TODO: catch error type and display correct message
-                    self.navigationVC?.addStatusBar(text: .apiError)
-                    self.diningInsights = DiningAPI.instance.getCachedDiningInsights()
-                }
-                self.updateHostingView()
+            switch result {
+            case .success(let diningInsights):
+                self.diningInsights = diningInsights
+            case .failure(.authenticationError):
+                //TODO: handle authenticationError
+                print("reauthentication?")
+            case .failure(.noInternet):
+                DispatchQueue.main.async { self.navigationVC?.addStatusBar(text: .noInternet) }
+            default:
+                DispatchQueue.main.async { self.navigationVC?.addStatusBar(text: .apiError) }
             }
+            
+            self.diningInsights = DiningAPI.instance.getCachedDiningInsights()
```

Comment:
can we do this only when there is a failure? it's a little confusing that we do this all the time (even though it works). we should use the result if it was .success.

Reply:
I thought that the cache exist such that even if a user, for example, opens their dining insights tab without the internet, if they had access to the internet and cached the result of their dining insights previously, they would still be able to view it.

Reply:
resolved